cryptoki/src/lib.rs

```diff
@@ -70,6 +75,7 @@ pub struct Pkcs11 {
     #[derivative(Debug = "ignore")]
     _pkcs11_lib: cryptoki_sys::Pkcs11,
     function_list: cryptoki_sys::_CK_FUNCTION_LIST,
+    management_sessions: ManagementSessionMap,
```

Comment:
I think you should add logic in `Drop`, below, to close any outstanding sessions before finalizing (unless you think it doesn't make any difference?)

Reply:
YES!  Forgot to do that!

Reply:
Technically it shouldn't make a difference, calling Finalize should close all sessions anyways, but that would be cleaner for sure.